library/src/main/kotlin/io/intrepid/bleidiom/BleIdiomImpl.kt

```diff
@@ -174,10 +178,10 @@ internal class BleCharValueDelegate<Val : Any>() : BleCharHandlerDSL<Val> {
             letMany(uuid, service?.sharedConnection, fromByteArray) { charUUID, connection, transform ->
                 connection.firstOrError().flatMapTry {
                     it.readCharacteristic(charUUID)
-                            .toRx2()
-                            .singleOrError()
-                            .map { byteArray -> transform(byteArray) }
-                            .doOnSuccess { backingField.currentValue = it }
+                        .toObservable()
```

Comment:
Not sure, but I think that `readCharacteristic(...)` returns a `Singe`. 
If so, remove `.toObservable().singleOrError()` entirely.